File: Mobile Applications/Intermediate/Currency_Converter/app/src/main/java/com/example/currencyconverter/MainActivity.java
package com.example.currencyconverter;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
  public void convertCurrency(View view) {
      Log.i("Info","Button pressed");

      EditText editText = (EditText) findViewById(R.id.editText);

      String amountInDollar = editText.getText().toString();
      Double amountInDollarDouble = Double.parseDouble(amountInDollar);

      double amountInPoundsDouble = amountInDollarDouble * 1.3;

      String amountInDollarsString = Double.toString(amountInDollarDouble);
      Log.i("Amount in Dollars", amountInDollarsString);

  }



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: Mobile Applications/Basic/Voicetotext/app/src/main/java/com/example/voicetotext/MainActivity.kt
@file:Suppress("DEPRECATION")

package com.example.voicetotext

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.speech.RecognizerIntent
import android.widget.Button
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.view.*
import org.w3c.dom.Text
import java.util.*
import kotlin.collections.ArrayList

class MainActivity : AppCompatActivity() {
    lateinit var btn : Button
    lateinit var txt : TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn = findViewById(R.id.btn)
        txt = findViewById(R.id.text)

        btn.setOnClickListener {
            val intent = Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH)
            intent.putExtra(
                RecognizerIntent.EXTRA_LANGUAGE_MODEL,
                RecognizerIntent.LANGUAGE_MODEL_FREE_FORM
            )
            intent.putExtra(RecognizerIntent.EXTRA_LANGUAGE, Locale.getDefault())

            intent.putExtra(RecognizerIntent.EXTRA_PROMPT, "Say Here")

            startActivityForResult(intent, 100)

        }
    }

         override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
            super.onActivityResult(requestCode, resultCode, data)

            if(requestCode == 100 || data != null){
                val res : ArrayList<String> = data!!.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS) as ArrayList<String>
                txt.text= res[0]
            }
        }
    }

